test: Component picker test is added

pomi opened the pull request:
## Changes
Component picker test is added

Commit:
Migration from cypress to playwright, updated

Commit:
Flow Container test is added

Commit:
preliminary results for component picker test

Commit:
Obsolete comments are deleted

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
Merge branch 'main' into AESQ-2883

Commit:
preliminary commit

Commit:
Merge branch 'pw_poc' into AESQ-3621

Commit:
Component picker test is added

Commit:
Files updates to comply with MR comments

Commit:
Files updates to comply with MR comments

Commit:
preliminary commit

Commit:
base page was updated to resolve conflicts

Commit:
fixed merge conflicts

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,147 @@
+  constructor(page: Page) {
+    super(page);
+    this.page = page;
+    this.flowContainerDefault = '#gatsby-focus-wrapper > div:nth-child(1) > div.my-2.container.mx-auto > div:nth-child(4) > section > div';
```
@pomi ,
 
the selectors have to be replaced with the stable ones. Please work with the developer to add the needed attributes to be used in the tests.

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+import { expect, Page, test } from '@playwright/test';
+import { FlowContainerPage } from '../../pages/flow-container-page';
+
+const NUMBER_OF_COMPONENTS = 26;
```
@pomi ,

1) please move it to FC PO
2) Please pay attention, that the total number of displayed components is 25. 
26th element is message 'This result set includes items which cannot be displayed. Please reduce the result set to less than 25 by filtering or searching.'

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+
+async function checkAllComponentCheckboxes(page: Page, flowContainerPage: FlowContainerPage) {
+  expect(await page.locator(flowContainerPage.component.richText).getAttribute('disabled')).toBeFalsy();
+  expect(await page.locator(flowContainerPage.component.richText).getAttribute('disabled')).toBeFalsy();
```
duplicates a step above

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+  test.beforeEach(async ({ browser }) => {
+    const context = await browser.newContext();
+    page = await context.newPage();
+    await page.goto('/flow-container');
```
a trailing slash is missing

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await flowContainerPage.toggleEditMode();
+    await page.click(flowContainerPage.flowContainerDefault);
+    await page.click(flowContainerPage.addFlowContainerButton);
+    expect.soft(await page.locator('#gridlistboxinner > div').count()).toEqual(NUMBER_OF_COMPONENTS);
```
@pomi ,

here and below in all test(), please move all element locators into FC PO

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    expect.soft(await page.locator(flowContainerPage.addComponentForm).isVisible()).toBeFalsy();
+  });
+
+  test.only('Component picker: 2 - checking Filter functionality - Tout', async () => {
```
@pomi 

1) .only  has to be deleted - it runs just this one test
2) replace Tout with Card

TatsianaMiraniuk commented on playwright/pages/flow-container-page.ts:
```diff
@@ -0,0 +1,147 @@
+      noBody: '#No\\ Body',
+      noTitleAndBody: '#No\\ Title\\ and\\ Body',
+    };
+    this.clearLink = 'body > div:nth-child(34) > div > div > div > div.rc-tooltip-inner > div > form > div > div > div.bl-pt-grid-2.w-1\\/4.overflow-y-auto > a';
```
@pomi ,

2nd test fails on this selector

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    expect.soft(await page.locator('#gridlistboxinner > div').count()).toEqual(16);
+
+    expect.soft(await page.locator(flowContainerPage.component.richText).getAttribute('disabled')).toEqual('');
+    expect.soft(await page.locator(flowContainerPage.component.richText).getAttribute('disabled')).toEqual('');
```
duplicated

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    expect.soft(await page.locator(flowContainerPage.typeSection).innerText()).toEqual('Type');
+    expect.soft(await page.locator(flowContainerPage.orientationSection).innerText()).toEqual('Orientation');
+
+    await flowContainerPage.typeText(flowContainerPage.searchField, 'Rich text');
```
@pomi ,

it's better to move checking Search into a separate test()

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await page.locator(flowContainerPage.searchField).fill('');
+    expect.soft(await page.locator('#gridlistboxinner > div').count()).toEqual(NUMBER_OF_COMPONENTS);
+
+    await page.locator(flowContainerPage.component.accordion).click();
```
it can be moved into a separate test()

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await page.locator(flowContainerPage.component.accordion).click();
+    expect.soft(await page.locator('#gridlistboxinner > div').count()).toEqual(NUMBER_OF_COMPONENTS);
+
+    await page.locator(flowContainerPage.component.image).click();
```
it can be moved into a separate test()

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await page.locator(flowContainerPage.component.image).click();
+    expect.soft(await page.locator('#gridlistboxinner > div').count()).toEqual(NUMBER_OF_COMPONENTS);
+
+    await page.locator(flowContainerPage.component.contentful).click();
```
it can be moved into a separate test()

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    expect.soft(await page.locator(flowContainerPage.orientation.NA).getAttribute('disabled')).toBeFalsy();
+    expect.soft(await page.locator(flowContainerPage.landscapeLinkableImage).isVisible()).toBeTruthy();
+    expect.soft(await page.locator(flowContainerPage.squareLinkableImage).isVisible()).toBeTruthy();
+    // expect.soft(await page.locator(flowContainerPage.squareImage).isVisible()).toBeTruthy();
```
that should be uncommented/tested too

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await page.click(flowContainerPage.closeFormButton);
+    expect.soft(await page.locator(flowContainerPage.addComponentForm).isVisible()).toBeFalsy();
+
+    await page.click(flowContainerPage.flowContainerDefault);
```
it can be moved into a separate test()

TatsianaMiraniuk commented on playwright/tests/smoke/componentPicker.spec.ts:
```diff
@@ -0,0 +1,197 @@
+    await page.close();
+  });
+
+  test('Component picker: 1 - checking presence of all components by amount', async () => {
```
@pomi ,

please split to a smaller test by a common sense depending on what is being tested

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
preliminary commit

Commit:
Merge branch 'pw_poc' into AESQ-2883

Commit:
Tests are updated according to PR comments

Commit:
Tests are updated according to PR comments

Commit:
Tests are updated according to PR comments

Commit:
Merge branch 'pw_poc' into AESQ-3621

Commit:
Merge branch 'pw_poc' into AESQ-3621

Commit:
feat(playwright): Migration from cypress to playwright (#1487)

* Migration from cypress to playwright

* Migration from cypress to playwright, updated

* Migration from cypress to playwright, updated

* typeText method was removed from pdp.spec.ts

* fix: add container selectors on flow-container page

* Use area-label insead of className

* FlowContainer component: allow id prop

* fixed all PR comments

* AESQ-8348, package-lock.json updated

* AESQ-8348, obsolete files were deleted

* AESQ-2883, viewport removed from all files

* AESQ-8348, obsolete files were deleted

* Preliminary commit

* AESQ-8348, fully reworked tests, 1-1 to cypress

* removed obsolete files

* Files updates to comply with MR comments

* Fixed according to MR

* Fixed according to MR

* Fixed according to MR

* Fixed viewport for chromium

* fixed tests according to PR comments

* preliminary commit

* Updates according to PR comments

* fix: add container selectors on flow-container page

* Use area-label insead of className

* FlowContainer component: allow id prop

* package-lock.json updated with playwright 1.22 version

* package-lock.json updated with playwright 1.22 version

* changed number of commits to 1 in commit history check

* limit number of workers for playwright

* cards.spec.ts was excluded from test suite

* fixed commit history comparison

Co-authored-by: pomi <AQ1sw@DE3fr$GT5>
Co-authored-by: tmiraniuk <tmiraniu@its.jnj.com>
Co-authored-by: Konstantin Mishur <kmishur@its.jnj.com>
Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

lucas-varela committed:
fix(editors): Fix hover menu closing on click (#1799)

* fix(editors): Fix hover menu closing on click

* Fix toggle trying to select an invalid range

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

wodenx committed:
fix(fclasses): Improve mobx warning. (#1794)

Commit:
fixed according to PR comments

misterjones committed:
docs(bodiless-documentation): Update release process (#1482)

* Add "Normal Release Process" section.
  * This includes Pre-Releases.
* Add "Fully Manual Release Process" section.

Co-authored-by: Chris Oden <coden@its.jnj.com>
Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

Commit:
fixed according to PR comments

misterjones committed:
docs(bodiless-schema-org, vital-card, vital-youtube): Vital Hero (#1792)

* Fix generation/build of documentation.
  * Add `bodiless-schema-org`, `vital-card`, and `vital-youtube`
    packages to Test Site and Vital Demo Site `package.json` files.
  * Add `getDocs.js` files to `bodiless-schema-org`, `vital-card`, and
    `vital-youtube` packages.
  * Update `package.json` files of `bodiless-schema-org`, `vital-card`,
    and `vital-youtube` to include `getDocs.js` and `@bodiless/cli`.
* Update doc structure
* Update `bodiless.docs.json` files.
  * Rename main doc file to `README.md`.
* Update package `README`s.
* Add `assets` folders with images.
* Update docs with Vital Hero info.
  * Update Vital Card, Image, and YouTube documentation with Hero
    details.
  * Clean up Bodiless YouTube Component page.
  * Update Schema page.
* Fix broken links in Vital Demo style guide.
* Break Product and Video Schema sections into pages
  * Break the "Product Schema" and "Video Schema" sections out of the
    main "Schema Component" page, and into their own pages under the
    Schema section.
* Fill out Vital Card page.
  * Add info block about Card height.

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>
Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

hvanyo committed:
feat(vital-navigation): Add Breadcrumbs (#1778)

* feat(vital-link): add buttons

* add breadcrumbs

* add breadcrumbs

* update buttons styles

* fix breadcrubm

* Update Buttons with hover

* update button

* fix spacing

* move buttons to vital-buttons

* add withBreadcrumbstore

* Move Link -> Buttons in styleguide

* fix nodekeys

* Working buttons

* move WTB separate file

* Rename to use ButtonClean/asButtonToken

* fix lint error

* fix shadowing

* fix vitalBreadcrumbs.Default

* fix link size

* remove button

* remove more buttons changes

* remove lint error

* add TextPrimaryInteractiveHover

* fix(vital-navigation): missing node collection for breadcrumb

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong

* fix(vital-navigation): For a non-Parent menu with submenu items -- breadcrumb link is wrong; FinalTrail styling fix

* fix(vital-navigation): last item bold

* fix(vital-navigation): last item bold

* fix(vital-navigation): last item bold

Co-authored-by: Dewen Li <dli94@its.jnj.com>

hvanyo committed:
feat(vital-buttons): Add Buttons (#1781)

* feat(vital-buttons): add buttons


Co-authored-by: Dewen Li <dewen.sinocrest@gmail.com>

Asemirski committed:
fix: checkout to merge commit on pull_request_target event (#1808)

Commit:
Merge branch 'main' into AESQ-2883

lucas-varela committed:
fix(lists): Fix list context menu not opening on tab (#1771)

* Fix list context menu not opening on tab

* Activate list item's context on creation

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

lucas-varela committed:
feat(search): Add @bodiless/vital-search (#1628)

* WIP: add vital-search package

* Fix vital-search dependency versions

* Update vital-search git url

* Update vital-search docs config file

* Add bodiless search reexports into vital search

* Remove vital-layout dependency from vital-search

* Update search box styles

* Add search page to vital-demo

* Update search configuration lookup

* Update vital-search docs

* Update vital-search styles

* Update packages to 1.0.0-beta.12

* Add minor style fixes

* Fix typo in bodiless-search docs

* Add mobile styles to the search component

* Update vital search dependencies

* Move search components into vital-search

* Add copyright message to vital-serch files

* Update vital-search version

* Refactor vital-search component structure

* Update vital-search imports on vital-layout

* Rename SearchResult to SearchResults

* Rename SearchResultItem to SearchResult

* Update SearchLayoutComponents type

* Move search context to a separate layout token

* Update vital-search tokens

* Update Search page template structure

* Remove ResponsiveBreakpoints helper

* Update vital-search to 1.0.0-beta.18

* Update vital-search token exports

* Fix __vital__ site build breaking

* Remove styleguide pages from vital-demo search

* Add H1Link through H5Link typography tokens

* Fix homepage search result not having a title

* Add colored backgrounds to search suggestion rows

* Fix SearchTool test

* Fix broken home page href in search results

* Fix vital-search colors missing on static sites

* Update vital-search to v1.0.0-rc.1

* Add missing search secondary color token

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

hvanyo committed:
fix(vital-cards): update cards with buttons & misc fixes (#1805)

* Fix set of issues with cards/buttons & other vital design issues.

Commit:
Merge branch 'main' into AESQ-3621

Commit:
Merge branch 'main' into AESQ-2883

Commit:
test: smoke tests for Flow Container (#1809)

* smoke test for flow containers

Commit:
fixed merge conflicts

misterjones committed:
docs(bodiless-documentation, bodiless-psh): Add Netlify onboarding (#1813)

* Add Netlify onboarding documentation.
* Create "Web Platforms" section in documentation.
  * Netlify page is here.
  * Move Platform.sh page here.

Commit:
Merge branch 'AESQ-2883' into AESQ-3621

# Conflicts:
#	playwright/pages/base-page.ts
#	playwright/pages/flow-container-page.ts

Commit:
fixed merge conflicts

Commit:
Merge branch 'main' into AESQ-3621

misterjones committed:
docs(vital-buttons): Update Vital Buttons docs (#1815)

* Update "Vital Button Component" page.
  * Add screenshots.
  * Rename main doc to `README.md`.
* Update package README.
* Update Fontello font docs

Co-authored-by: Heidi Vanyo <58534559+hvanyo@users.noreply.github.com>

Commit:
Merge branch 'main' into AESQ-3621

hvanyo committed:
fix(vital-test, __vital__): fix Shadowing and move to vital-test (#1787)

* addressed shadowing bugs 
* Added vital test package

marcopagliarulo committed:
feat(bodiless-core): BodilessJS Remove AsBodilessComponent Edit code from Static bundle #1542 (#1785)

* feat(bodiless-core): Use static replacement on asBodilessComponent

* Refactor bodiless hydration

* Fix missing wrapper on static page

* fix missing withPreview on static

* Create static version of richtext

* Gatsby page static

* remove gatsbyLink from test-site editor

* Add exclude option to staticReplacementPlugin

* Enable StaticReplacementPlugin on test site

* Fix richtext tests, add static richtext test, remove not relevant test case from hydration

* Remove gatsby link from test-site

* Address feedbacks

* fix unit tests

* Add RichTextStatic test snapshot

* restore missing export

* fix component disappearing when parent is re rendered before withoutHydration get hydrated

* Remove gatsby-link dependency from test-site

Co-authored-by: Marco Pagliarulo <mpaglia1@its.jnj.com>

Commit:
Merge branch 'main' into AESQ-3621
